File: components/jpa-handling-session/src/main/java/com/sv/youapp/component/authorization/entities/jpa/GrantTypeEntity.java
package com.sv.youapp.component.authorization.entities.jpa;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Entity
@Table(name = "grant_type")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class GrantTypeEntity {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@Column(nullable = false, unique = true)
	private String name;

	@ManyToMany(mappedBy = "grantTypes")
	private Set<RegisteredClientEntity> clients;
}

File: components/jpa-handling-session/src/main/kotlin/com/sv/youapp/component/authorization/entities/jpa/GrantTypeEntity.kt
package com.sv.youapp.component.authorization.entities.jpa

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table

@Entity
@Table(name = "grant_type")
class GrantTypeEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int,
    @Column(nullable = false, unique = true)
    var name: String,
) {
    @ManyToMany(mappedBy = "grantTypes")
    lateinit var clients: MutableSet<RegisteredClientEntity>
}
